File: c++/fib.cpp
#include<vector>
#include<iostream>
#include<string>

using namespace std;

vector<int> fib(int count);

int main(int argc, char** argv) {
  string count = argv[1];
  auto v = fib(stoi(count));
  for (auto n: v)
    cout << n << endl;
  return 0;
}

vector<int> fib(int count) {
  vector<int> v;
  int x = 1;
  int y = 1;
  for(int i = 0; i < count; i++) {
    int tmp = x;
    v.push_back(x);
    x = y;
    y = y + tmp;
  }
  return v;
}

File: rust/fib.rs
use std::env;

fn main() {
    let args: Vec<String> = env::args().collect();
    let count: i32 = args[1].parse::<i32>().unwrap();
    for i in fib(count).iter() {
        println!("{}", i);
    }
}

fn fib(n: i32) -> Vec<i32> {
    let mut a = 1;
    let mut b = 1;
    let mut v = Vec::new();

    for _i in 0..n {
        v.push(a);
        let tmp = a;
        a = b;
        b = b + tmp;
    }
    return v;
}

